commands/build-image.ts

```diff
@@ -95,20 +93,28 @@ export async function buildImage(actionContext: IActionContext, dockerFileUri?:
     if (defaultContextPath && defaultContextPath !== '') {
         contextPath = defaultContextPath;
     }
+    let absFilePath: string = path.join(vscode.workspace.workspaceFolders[0].uri.fsPath, dockerFileItem.relativeFolderPath, dockerFileItem.relativeFilePath);
+    let dockerFileKey = buildImageNameFromPath(absFilePath);
+    let prevImageName: string | undefined = ext.context.globalState.get(dockerFileKey);
+    let imageName: string;
 
     // Get imageName based on name of subfolder containing the Dockerfile, or else workspacefolder
-    let imageName: string;
-    imageName = path.basename(dockerFileItem.relativeFolderPath).toLowerCase();
+    imageName = prevImageName || path.basename(dockerFileItem.relativeFolderPath).toLowerCase();
+
     if (imageName === '.') {
         imageName = path.basename(rootFolder.uri.fsPath).toLowerCase();
     }
-
-    const imageWithTag: string = await getTagFromUserInput(imageName + ":latest");
-    const defaultPath = configOptions.get(configurationKeys.defaultRegistryPath, '');
-    if (!imageWithTag.includes(defaultPath)) { //user has entered a prefix different from stored
-        await askToSavePrefix(imageWithTag);
+    if (!imageName.match(/:[^\/]+$/)) {
+        imageName += ":latest"
     }
+
+    const imageWithTag: string = await getTagFromUserInput(imageName);
```

Comment:
We don't want to use getTagFromUserInput, because it's adding the default registry path (which we only want for the first build) and it's highlighting the registry path, which I don't think makes sense for build.